ci/Jenkinsfile

```diff
@@ -0,0 +1,116 @@
+@Library('gpuci_shared_lib') _
+
+pipeline {
+  agent any
+  options {
+    buildDiscarder(logRotator(numToKeepStr: '10'))
+    ansiColor('xterm')
+  }
+  stages {
+    stage("Style Checks") {
+      steps {
+        sh "env"
+      }
+    }
+    stage("C++ Matrix Build") {
+      environment {
+        BUILD_MODE = 'pull-request'
+        PARALLEL_LEVEL = '4'
+        CUDA="11.5"
+      }
+      steps {
+        script {
+          parallel generateStage("cuda_build", {
+            checkout scm
```

Comment:
It looks like this block is missing a `cleanWs` call. However, we should move all `cleanWs` and `checkout scm` calls to the shared library, since they will likely be used by every repository.